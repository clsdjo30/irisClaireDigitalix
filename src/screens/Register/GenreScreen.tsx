import React, { useState } from 'react';
import { StyleSheet, View, Text, Dimensions } from 'react-native';
import { Icon } from '@rneui/base'
import { StackScreenProps } from '@react-navigation/stack';
import { useUserStore } from '../../hooks/useUserStore';
import { colors } from '../../theme'
import NavigationButton from '../../components/NavigationButton';
import { CheckBox } from '@rneui/themed';

const width = Dimensions.get('window').width;



const GenreScreen: React.FC<StackScreenProps<any>> = ({ navigation }) => {
    const [user, setUser] = useUserStore()
    const [error, setError] = useState('')
    const [homme, setHomme] = useState(false)
    const [femme, setFemme] = useState(false)
    const [autres, setAutres] = useState(false)

    const resetAll = () => {
        setHomme(false);
        setFemme(false);
        setAutres(false);
    }

    const handleHommeCheck = () => {
        resetAll();
        setHomme(true);
        setUser({ ...user, genre: 'homme' })
    }

    const handleFemmeCheck = () => {
        resetAll();
        setFemme(true);
        setUser({ ...user, genre: 'femme' })
    }

    const handleAutresCheck = () => {
        resetAll();
        setAutres(true);
        setUser({ ...user, genre: 'autres' })
    }

    function goToBirthday() {
        if (user.genre.length === 0) {
            setError("Vous devez choisir un genre");
            return;
        }

        navigation.navigate('Birthday');
    }


    return (
        <View testID='genre-screen' style={styles.container}>
            {error !== '' && (
                <View style={styles.errorContainer}>
                    <Text style={styles.errorText}>{error}</Text>
                </View>
            )}
            <View style={styles.controls}>
                <View style={styles.genderTitle}>
                    <Text style={styles.contentTitle}>Vous Ãªtes : </Text>
                </View>
                <View style={styles.checkContainer}>
                    <CheckBox
                        title="Un Homme"
                        size={20}
                        checked={homme}
                        checkedTitle='Je suis un homme'
                        checkedIcon={
                            <Icon
                                name="man-outline"
                                type="ionicon"
                                color={colors.palette.purple400}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        uncheckedIcon={
                            <Icon
                                name="man-outline"
                                type="ionicon"
                                color={colors.palette.white}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        onPress={handleHommeCheck}
                        containerStyle={styles.genderCheckBox}
                        wrapperStyle={homme ? styles.positionCheckedBox : styles.positionBox}
                        titleProps={homme ? { style: styles.checkedText } : { style: styles.checkText }}

                    />
                    <CheckBox
                        title="Une Femme"
                        checked={femme}
                        checkedTitle='Je suis une femme'
                        checkedIcon={
                            <Icon
                                name="woman-outline"
                                type="ionicon"
                                color={colors.palette.purple400}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        uncheckedIcon={
                            <Icon
                                name="woman-outline"
                                type="ionicon"
                                color={colors.palette.white}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        onPress={handleFemmeCheck}
                        containerStyle={styles.genderCheckBox}
                        wrapperStyle={femme ? styles.positionCheckedBox : styles.positionBox}
                        titleProps={femme ? { style: styles.checkedText } : { style: styles.checkText }}
                    />
                    <CheckBox
                        title="Autres"
                        checked={autres}
                        checkedTitle='Je ne sais pas'
                        checkedIcon={
                            <Icon
                                name="transgender-outline"
                                type="ionicon"
                                color={colors.palette.purple400}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        uncheckedIcon={
                            <Icon
                                name="transgender-outline"
                                type="ionicon"
                                color={colors.palette.white}
                                size={30}
                                iconStyle={{ marginRight: 10 }}
                            />
                        }
                        onPress={handleAutresCheck}
                        containerStyle={styles.genderCheckBox}
                        wrapperStyle={autres ? styles.positionCheckedBox : styles.positionBox}
                        titleProps={autres ? { style: styles.checkedText } : { style: styles.checkText }}
                    />

                </View>
            </View>
            <NavigationButton
                color={colors.palette.violetBg}
                backgroundColor={colors.palette.orange}
                width={width * 0.85}
                title="Suivant"
                onPress={goToBirthday}
            />
        </View >
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: colors.palette.purple600,
    },
    icon: {
        marginRight: 16,
    },
    genderTitle: {
        width: width * 0.8,
        flexDirection: 'row',
        justifyContent: 'flex-start',
        alignItems: 'center',
    },

    //CHECKBOX
    checkContainer: {
        width: width * 0.8,
        flexDirection: 'column',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginTop: 10,
        marginBottom: 10,
    },
    genderCheckBox: {
        backgroundColor: colors.palette.violet,
    },
    checkText: {
        color: colors.palette.white,
        fontSize: 20,
        fontFamily: 'mulishLight',
        marginLeft: 20,
    },
    checkedText: {
        color: colors.palette.purple400,
        fontSize: 20,
        fontFamily: 'mulishSemiBold',
        marginLeft: 20,
    },
    positionBox: {
        width: width * 0.8,
        backgroundColor: colors.palette.purple500,
        borderRadius: 8,
        paddingVertical: 10,
        paddingHorizontal: 20,
    },
    positionCheckedBox: {
        width: width * 0.8,
        backgroundColor: colors.palette.purple100,
        borderRadius: 8,
        paddingVertical: 10,
        paddingHorizontal: 20,
    },
    controls: {
        flexDirection: 'column',
        alignItems: 'center',
        marginBottom: 40,
    },
    contentTitle: {
        fontFamily: "mulishSemiBold",
        fontSize: 20,
        color: colors.palette.violetClair,
        marginBottom: 20
    },
    //ERROR
    errorContainer: {
        position: 'absolute',
        top: 0,
        width: '80%',
        backgroundColor: colors.palette.orange,
        marginTop: 40,
        borderRadius: 16,
    },
    errorText: {
        textAlign: "center",
        fontFamily: "mulishBold",
        fontSize: 12,
        color: colors.palette.violetClair,
        paddingVertical: 6,
    },
});

export default GenreScreen;