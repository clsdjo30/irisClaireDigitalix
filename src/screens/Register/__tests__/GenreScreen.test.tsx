import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import GenreScreen from '../GenreScreen';

// Mock the navigation object with required methods
const mockNavigation: any = {
    navigate: jest.fn(),
};
const mockRoute: any = {
    key: 'MockKey',
    name: 'MockName',
};

// Mock the Icon component from react-native-vector-icons/FontAwesome
jest.mock('@rneui/base', () => ({
    Icon: () => 'Icon',
}));

describe('GenreScreen', () => {
    it('renders correctly and matches the snapshot', () => {
        const tree = render(<GenreScreen navigation={mockNavigation} route={mockRoute} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const { getByTestId } = render(<GenreScreen navigation = { mockNavigation } route = { mockRoute }  />);
        expect(getByTestId('genre-screen')).toBeTruthy();
    });

    it('contains the key elements', () => {
        const { getByText} = render(<GenreScreen navigation={mockNavigation} route={mockRoute} />);
        expect(getByText('Vous Ãªtes : ')).toBeTruthy();
        expect(getByText('Choisissez votre genre')).toBeTruthy();
    });

    it('navigates to the next screen when the Next button is pressed', () => {
        const { getByText } = render(<GenreScreen navigation={mockNavigation} route={mockRoute} />);
        const nextButton = getByText('Suivant');
        fireEvent.press(nextButton);
        expect(mockNavigation.navigate).toHaveBeenCalledWith('Birthday', { user: expect.any(Object) });
    });
});
