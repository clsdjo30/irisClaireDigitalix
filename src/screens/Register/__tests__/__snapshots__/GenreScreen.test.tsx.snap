// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreScreen renders correctly and matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2F2A5A",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
    }
  }
  testID="genre-screen"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "marginTop": 40,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "width": 600,
        }
      }
    >
      <Text
        style={
          {
            "color": "#eeedf6",
            "fontFamily": "mulishRegular",
            "fontSize": 18,
            "marginBottom": 20,
          }
        }
      >
        Vous Ãªtes : 
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "width": 637.5,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "space-between",
              "marginBottom": 24,
              "paddingLeft": 8,
            },
            {
              "alignItems": "center",
              "backgroundColor": "#EEEDF6",
              "borderBottomColor": "#CBA135",
              "borderBottomWidth": 1,
              "borderLeftColor": "#CBA135",
              "borderLeftWidth": 1,
              "flexDirection": "row",
              "height": 60,
              "marginBottom": 20,
              "width": 637.5,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "maxWidth": "90%",
                "opacity": 1,
              },
              {
                "color": "#B7B7E1",
                "fontFamily": "mulishRegular",
                "fontSize": 14,
                "marginLeft": 14,
              },
            ]
          }
        >
          Choisissez votre genre
        </Text>
        <View
          style={
            [
              {
                "alignItems": "stretch",
                "alignSelf": "stretch",
                "backgroundColor": "white",
                "borderBottomRightRadius": 4,
                "borderTopRightRadius": 4,
                "justifyContent": "center",
                "marginRight": 8,
                "padding": 8,
              },
              {
                "backgroundColor": "#EEEDF6",
              },
            ]
          }
        >
          Icon
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#D81F28",
              "borderRadius": 8,
              "height": 48,
              "justifyContent": "center",
              "marginVertical": 32,
            },
            {
              "opacity": 1,
            },
            {
              "backgroundColor": "#FF7648",
              "borderRadius": 16,
              "height": 30,
              "marginTop": 30,
              "paddingVertical": 5,
              "position": "relative",
              "width": 637.5,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              {
                "color": "#eeedf6",
                "fontFamily": "mulishBold",
                "fontSize": 14,
              },
            ]
          }
        >
          Suivant
        </Text>
      </View>
    </View>
  </View>
</View>
`;
